src/field-modules/my-logs/components/EditLog.vue

```diff
@@ -941,6 +937,49 @@ export default {
       return this.logs[this.currentLogIndex].images.data
         .filter(img => typeof img === 'string');
     },
+    logAreas() {
+      return this.logs[this.currentLogIndex].area.data
+        ? this.logs[this.currentLogIndex].area.data
+        : null;
+    },
+    /*
+    Assemble layers for display.
+    The 'previous' layer is assembled from the geofield plus
+    all area geometires associated with the log.
+    The 'movement' layer is the geometry in the log's movement field
+    */
+    mapLayers() {
+      const movement = {
+        title: 'movement',
+        wkt: this.logs[this.currentLogIndex].movement.data.geometry
+          ? this.logs[this.currentLogIndex].movement.data.geometry
+          : null,
+        color: 'orange',
+        visible: true,
+      };
+      const assembledPrevious = () => {
+        const fields = [];
+        if (this.logs[this.currentLogIndex].geofield.data.length > 0) {
+          fields.push(this.logs[this.currentLogIndex].geofield.data[0].geom);
+        }
+        this.logAreas.forEach((logArea) => {
+          const areaGeom = (this.areas.find(area => area.tid === logArea.id).geofield[0])
+            ? this.areas.find(area => area.tid === logArea.id).geofield[0].geom
+            : null;
+          if (areaGeom) { fields.push(areaGeom); }
+        });
+        return fields.length > 0
+          ? mergeGeometries(fields)
+          : null;
+      };
```

Comment:
Perhaps this would be more concise:

```js
      const previousGeoms = this.logAreas.map(logArea => (
        this.areas.find(area => area.tid === logArea.id).geofield?.[0].geom
      )).concat(this.logs[this.currentLogIndex].geofield.data?.[0]);
      const previousWKT = mergeGeometries(previousGeoms);
      const previous = {
        title: 'previous',
        wkt: previousWKT,
        color: 'blue',
        visible: true,
      };
```

I haven't tested that that works, but it seems a lot less noisy to me and easier to read. Note the use of [Optional Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).

Reply:
It doesn't work as written.  I'd definitely like to learn more about optional chaining, but I'm not sure I have time to dive into it today.

Reply:
Ah, ok. I would still encourage you to look towards ways you can chain [array "accessor" methods](https://developer.mozilla.org/tr/docs/Web/JavaScript/Reference/Global_Objects/Array/prototype#Accessor_methods), rather than mutating an array. This will make the code more declarative, less imperative, and I think easier to reason about. The optional chaining stuff is pretty straight-forward; basically, as the MDN docs describe:

> The `?.` operator functions similarly to the `.` chaining operator, except that instead of causing an error if a reference is nullish (null or undefined), the expression short-circuits with a return value of `undefined`.

That's it. Just a nice simple way to save some yourself time as well as the grief running null checks.